httputil/retry.go

```diff
@@ -178,7 +178,7 @@ func isDnsUnavailable(err error) bool {
 	// not exposed in net.DNSError and in go-1.10 it is not even
 	// a temporary error so there is no way to distiguish it other
 	// than a fugly string compare on a (potentially) localized string
-	return strings.Contains(dnsErr.Err, "Temporary failure in name resolution")
+	return strings.Contains(dnsErr.Err, "connection refused") || strings.Contains(dnsErr.Err, "Temporary failure in name resolution")
```

Comment:
With this change we now have the exact same conditional as the one inside `shouldRetryError()`; that means that with the re-ordered conditionals below we now treat this as a parmanent error and not an error that should be retried - therefore the network-retry spread test now fails. I'm investigating further how to untangle this, we basically need to make our mind here..